src/common/time/src/interval.rs

```diff
@@ -0,0 +1,708 @@
+    if is_time_interval_pos && is_only_time {
+        interval.push_str(&format!("{}:{:02}:{:02}", hours, mins, secs));
+    } else {
+        let minutes = mins.unsigned_abs();
+        let seconds = secs.unsigned_abs();
+        interval.push_str(&format!("{:+}:{:02}:{:02}", hours, minutes, seconds));
+    }
+    if micros != 0 {
+        let microseconds = format!(".{:06}", micros.unsigned_abs());
+        interval.push_str(&microseconds);
+    }
+    interval
+}
+
+/// IntervalCompare is used to compare two intervals
+/// It make interval into microseconds style.
+#[derive(PartialEq, Eq, Hash, PartialOrd, Ord)]
+struct IntervalCompare(i128);
+
+impl From<Interval> for IntervalCompare {
+    fn from(interval: Interval) -> Self {
+        Self(interval.to_micros())
+    }
+}
+
+impl Ord for Interval {
+    fn cmp(&self, other: &Self) -> Ordering {
+        IntervalCompare::from(*self).cmp(&IntervalCompare::from(*other))
+    }
+}
+
+impl PartialOrd for Interval {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Eq for Interval {}
+
+impl PartialEq for Interval {
+    fn eq(&self, other: &Self) -> bool {
+        self.cmp(other).is_eq()
+    }
+}
+
+impl Hash for Interval {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        IntervalCompare::from(*self).hash(state)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_interval_mul_int() {
+        let interval = Interval::new(1, 1, 1);
+        let interval2 = interval.checked_mul_int(2).unwrap();
+        assert_eq!(interval2.months, 2);
+        assert_eq!(interval2.days, 2);
+        assert_eq!(interval2.micros, 2);
+    }
+
+    #[test]
+    fn test_display() {
+        let interval = Interval::from_month_day_nano(1, 1, 1);
+        assert_eq!(interval.to_string(), "1 months 1 days");
+
+        let interval = Interval::from_month_day_nano(14, 31, 10000000000);
+        assert_eq!(interval.to_string(), "14 months 31 days 10000000 micros");
+    }
+
+    #[test]
+    fn test_interval_justified() {
+        let interval = Interval::from_month_day_nano(1, 131, 1).justified_interval();
+        let interval2 = Interval::from_month_day_nano(5, 11, 1);
+        assert_eq!(interval, interval2);
+
+        let interval =
+            Interval::from_month_day_nano(1, 1, 1000 * MICROS_PER_DAY * 2).justified_interval();
+        let interval2 = Interval::from_month_day_nano(1, 3, 0);
+        assert_eq!(interval, interval2);
+    }
+
+    #[test]
+    fn test_serde_json() {
+        let interval = Interval::new(1, 1, 1);
+        let json = serde_json::to_string(&interval).unwrap();
+        assert_eq!(
+            json,
+            "{\"months\":1,\"days\":1,\"micros\":1,\"unit\":\"MonthDayNano\"}"
+        );
+        let interval2: Interval = serde_json::from_str(&json).unwrap();
+        assert_eq!(interval, interval2);
+    }
+
+    #[test]
+    fn test_to_iso8601_string() {
```

Comment:
nitpick: Consider adding more test cases for `to_iso8601_string()`, `to_postgres_string()`, and `to_sql_standard_string()`